src/Paprika/Chain/Blockchain.cs

```diff
@@ -1145,6 +1176,118 @@ void ICommit.Visit(CommitAction action, TrieType type)
             }
         }
 
+        /// <summary>
+        /// Just for tests - accept visitor method to traverse the merkle trie
+        /// </summary>
+        /// <param name="visitor"></param>
+        /// <param name="path"></param>
+        /// <param name="context"></param>
+        public void Accept(IBlockstateVisitor visitor, NibblePath path, BlockstateVisitorContext context)
```

Comment:
Leaks all the Merkle construct back to the Blockchain. Should it be a method of `precommit` in total? So many cast downs here.

Reply:
Moved trie visitor to `ComputeMerkleBehavior` and removed from `Blockchain` component. Also exposed the visitor `Accept` method on `IRawState` and removed `TrieDumper` completely. This can be implemented in the NM client - `Nethermind.Paprika` or individual tests as it is only used there.